pallets/ajuna-affiliates/src/impls/fee_handler.rs

```diff
@@ -0,0 +1,35 @@
+use super::*;
+use ajuna_primitives::fee_handler::FeeProvider;
+use sp_runtime::{traits::CheckedDiv, Saturating};
+
+impl<T: Config<I>, I: 'static> FeeProvider for Pallet<T, I> {
+	type AccountId = AccountIdFor<T>;
+	type FeeIdentifier = RuleIdentifierFor<T, I>;
+	type FeeCurrency = BalanceOf<T, I>;
+	type FeeOutput = Vec<(Self::FeeCurrency, Self::AccountId)>;
+
+	fn get_fee_from(
+		base_fee: Self::FeeCurrency,
+		account: &Self::AccountId,
+		identifier: &Self::FeeIdentifier,
+	) -> Self::FeeOutput {
+		if let Some(chain) = Self::get_affiliator_chain_for(account) {
+			if let Some(rule_chain) = Self::get_rule_for(identifier) {
+				rule_chain
+					.into_iter()
+					.map(|rule_perc| {
+						base_fee
+							.saturating_mul(rule_perc.into())
+							.checked_div(&100_u32.into())
+							.unwrap_or_default()
+					})
+					.zip(chain)
+					.collect()
+			} else {
+				Vec::with_capacity(0)
+			}
+		} else {
+			Vec::with_capacity(0)
+		}
+	}
+}
```

Comment:
I am wondering if we should make this trait not depend on the balance type, but only return the fractions of the fee propagation. If we do it like that, the pallet will be independent of the currency. It is then the fee handlers responsibility to translate this in to actual balance values. What do you think about that?

Reply:
I think this approach is better since it is more clear who is responsible for calculating the fees, additionally with this approach we can indicate if no fee is to be paid for something wheres with the fractions you would still need to compute if the fraction leads to paying any fees or not at the outer level.

Reply:
Yes, I see your point. However, my arguments would be:

1. It would make sense to me if the instance computing the actual fee is the fee handler, instead of the pallet itself.
2. The fee handler could also early return if the base fee is 0 and doesn't even call into this code, which is more intuitive to me instead of first having to look into the affiliate code, and then understand the code, and then see, ah, it just returns a `Vec<(0, Beneficifiary>`.

I still tend towards moving the fee calculation into the fee handler for better separation of concerns, IMO the pallet should not care about how the distribution is interpreted, but we can also do this in a second step. 👍

Now, that I write this, I think we should definitely do the early return anyhow for readability and efficiency :D. Hence, implement my suggestion from 2.

Reply:
Hm, I agree to look into this again in step 2